addons/website_sale/controllers/main.py

```diff
@@ -1241,11 +1241,16 @@ def shop_address_submit(
         elif address_type == 'delivery':
             partner_fnames.add('partner_shipping_id')
 
+        previous_pricelist = order_sudo.pricelist_id
```

Comment:
Seems wrong you can prevent change pricelist from `_update_address` method instead first update pricelist and then set original again and recompute prices both time :thinking: